### PR TITLE
[FLINK-2769] [runtime-web] Add configurable job manager address to HTTP requests

### PR DESCRIPTION
**Pull request by uce**
These changes (re-)enable testing of changes to the web frontend:

- Set the `allow-origin` header in order to allow requests from the test server (`localhost:3000` requests to `localhost:8081`). I don't know how this was working before? We never set that before.
- Add job manager address to get requests and set default to empty String. This was completely removed in 3b8b4f0f8c0600dc851d676ce1bd7f5ab81cb64f, but it should have been only set to the empty string.

All `$http.get` requests should refer to the `jobServer` variable.

```
grep -r "\$http.get" flink-runtime-web/web-dashboard/app/
flink-runtime-web/web-dashboard/app//scripts/common/services.coffee:    $http.get flinkConfig.jobServer + "config"
flink-runtime-web/web-dashboard/app//scripts/modules/jobmanager/jobmanager.svc.coffee:    $http.get(flinkConfig.jobServer + "jobmanager/config")
flink-runtime-web/web-dashboard/app//scripts/modules/jobmanager/jobmanager.svc.coffee:    $http.get(flinkConfig.jobServer + "jobmanager/log")
flink-runtime-web/web-dashboard/app//scripts/modules/jobmanager/jobmanager.svc.coffee:    $http.get(flinkConfig.jobServer + "jobmanager/stdout")
flink-runtime-web/web-dashboard/app//scripts/modules/jobs/jobs.svc.coffee:    $http.get flinkConfig.jobServer + "joboverview"
flink-runtime-web/web-dashboard/app//scripts/modules/jobs/jobs.svc.coffee:    $http.get flinkConfig.jobServer + "jobs/" + jobid
flink-runtime-web/web-dashboard/app//scripts/modules/jobs/jobs.svc.coffee:      $http.get flinkConfig.jobServer + "jobs/" + jobid + "/config"
flink-runtime-web/web-dashboard/app//scripts/modules/jobs/jobs.svc.coffee:      $http.get flinkConfig.jobServer + "jobs/" + currentJob.jid + "/vertices/" + vertexid + "/subtasktimes"
flink-runtime-web/web-dashboard/app//scripts/modules/jobs/jobs.svc.coffee:      $http.get flinkConfig.jobServer + "jobs/" + currentJob.jid + "/vertices/" + vertexid
flink-runtime-web/web-dashboard/app//scripts/modules/jobs/jobs.svc.coffee:      $http.get flinkConfig.jobServer + "jobs/" + currentJob.jid + "/vertices/" + vertexid + "/accumulators"
flink-runtime-web/web-dashboard/app//scripts/modules/jobs/jobs.svc.coffee:        $http.get flinkConfig.jobServer + "jobs/" + currentJob.jid + "/vertices/" + vertexid + "/subtasks/accumulators"
flink-runtime-web/web-dashboard/app//scripts/modules/jobs/jobs.svc.coffee:      $http.get flinkConfig.jobServer + "jobs/" + currentJob.jid + "/exceptions"
flink-runtime-web/web-dashboard/app//scripts/modules/jobs/jobs.svc.coffee:    $http.get flinkConfig.jobServer + "jobs/" + jobid + "/yarn-cancel"
flink-runtime-web/web-dashboard/app//scripts/modules/overview/overview.svc.coffee:    $http.get(flinkConfig.jobServer + "/overview")
flink-runtime-web/web-dashboard/app//scripts/modules/taskmanager/taskmanager.svc.coffee:    $http.get(flinkConfig.jobServer + "taskmanagers")
flink-runtime-web/web-dashboard/app//scripts/modules/taskmanager/taskmanager.svc.coffee:    $http.get(flinkConfig.jobServer + "taskmanagers/" + taskmanagerid)
```

---

**If you can find the time, make sure to test this by really running the frontend and not only by looking at the diffs. We don't have test coverage for these parts...**

### CODE REVIEW

**Comment by mxm** [comment]
Wrong JIRA issue?

**Comment by uce** [comment]
Thanks! It was 2769 and not 2796. Updated the PR.

File: flink-runtime-web/web-dashboard/app/scripts/modules/overview/overview.svc.coffee [hunk 1]
```diff
@@ -24,7 +24,7 @@ angular.module('flinkApp')
   @loadOverview = ->
     deferred = $q.defer()
 
-    $http.get("overview")
+    $http.get(flinkConfig.jobServer + "/overview")
```
Comment (rmetzger) [inline flink-runtime-web/web-dashboard/app/scripts/modules/overview/overview.svc.coffee]: absolute url's (starting with '/') will break the YARN proxy stuff.

Comment (uce) [inline flink-runtime-web/web-dashboard/app/scripts/modules/overview/overview.svc.coffee]: Good catch. Was not on purpose

**Comment by sachingoel0101** [comment]
Tested it out locally. Works well with `gulp watch` and custom web frontend port.
A few things I noticed:
1. Job Manager's log and stdout are not accessible with CORS, since they're served via `StaticFileServerHandler` which doesn't set CORS headers. This however shouldn't be an issue. We need to have a better logs service anyway, for both Job manager and task managers IMO.
2. At some point, we will remove the `GET /jobs/<job-id>/yarn-cancel` method, which is only there due to a limitation of yarn. To this end, there needs to be an `OPTIONS` handlers for the `DELETE` request to send CORS response headers in the preflight phase. https://developer.mozilla.org/en-US/docs/Web/HTTP/Access_control_CORS#Preflighted_requests. Not for the foreseeable future though. :-')

**Comment by uce** [comment]
Thanks for the reviews Robert and Sachin!

I've addressed Robert's comment. If there are no objections, I would like to merge this.

**Comment by rmetzger** [comment]
I think its good to merge, +1

**Pull request closed without merging**